google_gke/cluster.tf

```diff
@@ -0,0 +1,212 @@
+      enable_resource_consumption_metering = resource_usage_export_config.value.enable_resource_consumption_metering
+      bigquery_destination {
+        dataset_id = resource_usage_export_config.value.dataset_id
+      }
+    }
+  }
+
+  # Add-Ons
+  addons_config {
+    gce_persistent_disk_csi_driver_config {
+      enabled = true
+    }
+
+    network_policy_config {
+      disabled = false
+    }
+  }
+
+  # Maintenance
+  maintenance_policy {
+    daily_maintenance_window {
+      start_time = var.maintenance_start_time
+    }
+
+    dynamic "maintenance_exclusion" {
+      for_each = var.maintenance_exclusions
+      content {
+        exclusion_name = maintenance_exclusion.value.name
+        start_time     = maintenance_exclusion.value.start_time
+        end_time       = maintenance_exclusion.value.end_time
+      }
+    }
+  }
+
+  # Configurations for Cluster's Default Nodepool / Nodes Defaults
+  # We fully expect the default node pool to be immediately removed.
+  default_max_pods_per_node = 32
+  remove_default_node_pool  = true
+
+  node_pool {
+    name               = "default-pool"
+    initial_node_count = 1
+
+    node_config {
+      labels = local.labels
+      tags   = local.tags
+    }
+  }
+
+  lifecycle {
+    ignore_changes = [
+      initial_node_count,
+      node_pool,
+      resource_labels["asmv"],
+      resource_labels["mesh_id"]
+    ]
+
+    prevent_destroy = true
+  }
+}
+
+#
+# GKE Node Pools as configured via Variables
+#
+resource "google_container_node_pool" "pools" {
+  for_each = local.node_pools
+  provider = google-beta
+  name     = each.key
+
+  cluster            = google_container_cluster.primary.name
+  initial_node_count = each.value.initial_node_count
+  location           = var.region
+  max_pods_per_node  = each.value.max_pods_per_node
+  project            = local.project_id
+
+  autoscaling {
+    min_node_count = each.value.min_count
+    max_node_count = each.value.max_count
+  }
+
+  management {
+    auto_repair  = true
+    auto_upgrade = true
+  }
+
+  node_config {
+    disk_size_gb = each.value.disk_size_gb
+    image_type   = "COS_CONTAINERD"
+    labels       = local.node_pools_labels[each.key]
+    dynamic "linux_node_config" {
+      for_each = length(local.node_pools_sysctls[each.key]) != 0 ? [1] : []
+
+      content {
+        sysctls = local.node_pools_sysctls[each.key]
+      }
+    }
+    machine_type    = each.value.machine_type
+    oauth_scopes    = local.node_pools_oauth_scopes[each.key]
+    service_account = google_service_account.cluster_service_account.email
+    tags            = local.node_pools_tags[each.key]
+    workload_metadata_config {
+      mode = "GKE_METADATA"
+    }
+  }
+
+  upgrade_settings {
+    max_surge       = each.value.max_surge
+    max_unavailable = each.value.max_unavailable
+  }
+
+  lifecycle {
+    create_before_destroy = true
+  }
```

Comment:
In my testing using an example like `complex1`, I needed to add the following, otherwise the node pool would try to be re-created:
```
    ignore_changes = [
      node_config[0].oauth_scopes,
      node_config[0].metadata,
    ]
```

Reply:
cool thanks will add.